### PR TITLE
web3 middleware

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/web3/web3.js/issues/6864
https://github.com/web3/web3.js/issues/6963

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by luu-alex — APPROVED** [review]

File: packages/web3-core/src/web3_request_manager.ts
Comment (avkos) [inline packages/web3-core/src/web3_request_manager.ts:173]: it is better to have brackets `{` and `}` even if it is very simple `if`. because in the future when someone extends this code they can make a mistake

File: tools/web3-plugin-example/src/custom_rpc_methods.ts
Comment (avkos) [inline tools/web3-plugin-example/src/custom_rpc_methods.ts]: Could you please check the prettier config on your IDE? Looks like it is not correct code automatically

**Review by avkos — CHANGES_REQUESTED** [review]

**Review by Muhammad-Altabba — APPROVED** [review]
Seems great! beside Oleksii's points.

However, I think it would be the best to use this feature in a plugin before we make it available in the next release. In order to see it in action and see if we need some changes that we can do without casing a breaking changes in the subsequent releases. So, can we for example set it as experimental till we try it inside some real case? And later we mark it as a read-to-use feature?

**Comment by jdevcs** [comment]
> Seems great! beside Oleksii's points.
> 
> However, I think it would be the best to use this feature in a plugin before we make it available in the next release. In order to see it in action and see if we need some changes that we can do without casing a breaking changes in the subsequent releases. So, can we for example set it as experimental till we try it inside some real case? And later we mark it as a read-to-use feature?

I also thought same to not include this in next release and test it with [Swisstronik usecase](https://www.npmjs.com/package/@swisstronik/web3-plugin-swisstronik), so in case any breaking changes required we can do before merging it ,I think we should merge this PR after next release ( and for next sprint's release ). 

I'll update and close implementation issue and created [next issue](https://github.com/web3/web3.js/issues/6963) for tracking above.

**Review by avkos — APPROVED** [review]

**Pull request merged**
